app/Entirety/semantics/prepDataSemantics.py

```diff
@@ -0,0 +1,134 @@
+import pandas as pd
+import json
+from projects.mixins import ProjectContextMixin
+from entities.requests import get_entities_list
+
+
+class PrepData(ProjectContextMixin):
+
+    def generate_df(self):
+        """
+        Generates a pandas DataFrame and a list of Cytoscape elements from a list of entities.
+
+        Returns:
+            list: A list of Cytoscape elements containing information about nodes and edges.
+                Each element is a dictionary with keys 'data' and 'classes', where 'data' is another dictionary
+                containing information about the node or edge, and 'classes' is a string specifying the type of the entity.
+        """
+        entity_list = get_entities_list(self, ".*", "", self.project)
+        entity_id_list = []
+        entity_type_list = []
+        entity_name_list = []
+        relationship_name_list = []
+        relationship_target_list = []
+
+        for entity in entity_list:
+            entity_json = json.loads(entity.json())
+
+            entity_id_list.append(entity.id)
+            entity_type_list.append(entity.type)
+
+            all_entity_values = self.all_values(entity_json)
+            entity_name = 'No name Set'  # default value if no "name" attribute is present
```

Comment:
@JRadebold @SBlechmann seems semantic app will require a "name" attribute. Is it the wanted behavior? Would it make more sense to show EntityID if no "name" is found?

Reply:
I see the option now. The implementation looks good!